common/common.cpp

```diff
@@ -90,6 +90,19 @@ void process_escapes(std::string& input) {
                 case '\'': input[output_idx++] = '\''; break;
                 case '\"': input[output_idx++] = '\"'; break;
                 case '\\': input[output_idx++] = '\\'; break;
+                case 'x':
+                    // Handle \x12, etc
+                    if (input_idx + 2 < input_len && input[input_idx + 1] != 0) {
+                        const char x[3] = { input[input_idx + 1], input[input_idx + 2], 0 };
+                        char *err_p = nullptr;
+                        const long val = std::strtol(x, &err_p, 16);
+                        if (*err_p == 0) {
```

Comment:
Well, the first reason is I didn't know to write it that way. :) I think there's a reason to keep it as-is though, creating a const 3 character array probably has lower overhead than making and destroying a `std::string`, concatenating some other strings, etc. It's probably a small difference, but hey, when I'm feeding a 100,000 token prompt to a 128K YaRN model maybe it'll be .5sec faster.

Reply:
Because of [small-string optimization](https://stackoverflow.com/a/10319672/21730044) there won't be any heap allocation, but I suppose std::string is still slightly more expensive.

If we keep the C version, we should replace `const char x[3]` with just `const char x[]` since the size is inferred.

C++ has had std::string_view and std::from_chars since 2017, but we can't use them :/

Reply:
I realized the previous code didn't correctly handle a case where the second hex digit was `\0`. This version is a little simpler and clearer also. However, I kept the `[3]` since I feel better about explicitly setting the size when doing pointer arithmetic on the array like that.